invoices/invoice_expiry_watcher.go

```diff
@@ -119,20 +118,54 @@ func (ew *InvoiceExpiryWatcher) AddInvoice(
 	}
 
 	expiry := invoice.CreationDate.Add(realExpiry)
-
-	log.Debugf("Adding invoice '%v' to expiry watcher, expiration: %v",
-		paymentHash, expiry)
-
-	newInvoiceExpiry := &invoiceExpiry{
+	return &invoiceExpiry{
 		PaymentHash: paymentHash,
 		Expiry:      expiry,
 	}
+}
+
+// AddInvoices adds multiple invoices to the InvoiceExpiryWatcher.
+func (ew *InvoiceExpiryWatcher) AddInvoices(
+	invoices []channeldb.InvoiceWithPaymentHash) {
+
+	var invoicesWithExpiry []*invoiceExpiry
+	for idx := range invoices {
+		newInvoiceExpiry := ew.prepareInvoice(
+			invoices[idx].PaymentHash, &invoices[idx].Invoice)
+		if newInvoiceExpiry != nil {
+			invoicesWithExpiry = append(invoicesWithExpiry, newInvoiceExpiry)
+		}
+	}
+
+	if len(invoicesWithExpiry) > 0 {
+		log.Debugf("Added %v invoices to the expiry watcher: %v",
+			len(invoicesWithExpiry))
+		select {
+		case ew.newInvoices <- invoicesWithExpiry:
+		case <-ew.quit:
+			// Select on quit too so that callers won't get blocked in case
```

Comment:
Typically we prefer to place comments like this right above the `case` statement.

Reply:
Sorry, was copy-paste. Done.